core/cache/metrics.go

```diff
@@ -93,6 +96,20 @@ func createControllerGauges() *ControllerGauges {
 				Help:      "The number of times the model config change hash was generated.",
 			},
 		),
+		CharmConfigHashCacheHit: prometheus.NewGauge(
+			prometheus.GaugeOpts{
+				Namespace: metricsNamespace,
+				Name:      "charm_hash_cache_hit",
```

Comment:
"charm_config_watcher_hash_hit"
"The number of times a change in master or branch config required no notification to config watcher(s)"

And the inverse for misses.